crates/configuration/src/shared/types.rs

```diff
@@ -27,6 +32,14 @@ impl From<u128> for U128 {
     }
 }
 
+// impl TryFrom<&str> for U128 {
+//     type Error = ConversionError;
+
+//     fn try_from(value: &str) -> Result<Self, Self::Error> {
+//         Ok(Self(value.to_string().parse::<u128>().unwrap()))
+//     }
+// }
+
```

Comment:
Why is is commented ? Could probably be reused below with some modification:

```suggestion
use std::error::Error;

impl TryFrom<&str> for U128 {
     type Error = Box<dyn Error>;

     fn try_from(value: &str) -> Result<Self, Self::Error> {
         Ok(Self(value.to_string().parse::<u128>()?))
     }
 }

```